Core/Core/Inbox/MessageDetails/View/MessageView.swift

```diff
@@ -23,32 +23,43 @@ public struct MessageView: View {
     @ScaledMetric private var uiScale: CGFloat = 1
 
     private var model: MessageViewModel
+    private let hideReplyButton: Bool
     private var replyDidTap: () -> Void
     private var moreDidTap: () -> Void
 
-    public init(model: MessageViewModel,
-                replyDidTap: @escaping () -> Void,
-                moreDidTap: @escaping () -> Void ) {
+    public init(
+        model: MessageViewModel,
+        hideReplyButton: Bool,
+        replyDidTap: @escaping () -> Void,
+        moreDidTap: @escaping () -> Void
+    ) {
         self.model = model
         self.replyDidTap = replyDidTap
         self.moreDidTap = moreDidTap
+        self.hideReplyButton = hideReplyButton
```

Comment:
Just my personal preference: if we always check the `hideReplyButton`'s negated value I think it would be better to save it as `showReplyButton` so we have to negate only inside the init and the conditions will be more straightforward to understand.